apollo-runtime-kotlin/src/iosMain/kotlin/com/apollographql/apollo/network/websocket/ApolloWebSocket.kt

```diff
@@ -0,0 +1,160 @@
+package com.apollographql.apollo.network.websocket
+
+import com.apollographql.apollo.ApolloWebSocketException
+import com.apollographql.apollo.network.toNSData
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.ReceiveChannel
+import okio.ByteString
+import okio.ByteString.Companion.toByteString
+import okio.IOException
+import okio.internal.commonAsUtf8ToByteArray
+import okio.toByteString
+import platform.Foundation.NSMutableURLRequest
+import platform.Foundation.NSOperationQueue
+import platform.Foundation.NSThread
+import platform.Foundation.NSURL
+import platform.Foundation.NSURLRequest
+import platform.Foundation.NSURLSession
+import platform.Foundation.NSURLSessionConfiguration
+import platform.Foundation.NSURLSessionWebSocketCloseCode
+import platform.Foundation.NSURLSessionWebSocketMessage
+import platform.Foundation.NSURLSessionWebSocketMessageTypeData
+import platform.Foundation.NSURLSessionWebSocketMessageTypeString
+import platform.Foundation.NSURLSessionWebSocketTask
+import platform.Foundation.setValue
+import kotlin.native.concurrent.AtomicReference
+
+typealias WebSocketFactory = (NSURLRequest) -> NSURLSessionWebSocketTask
+
+actual class ApolloWebSocketFactory(
```

Comment:
This is a big one: We need to make it clear that this will only work with iOS 13+ - I don't think this will even compile on older versions of iOS since it leverages web socket stuff introduced in iOS 13.

If we're not making iOS 13 a requirement for KMP (which we could, but I think might be a smidge premature), then there needs to be a way to prevent this from compiling or being available on iOS 12 and below.

Reply:
I was thinking on adding this as part of documentation later as we don't really promote new kotlin runtime module.

It's interesting is there any way to put restriction on iOS version in KN configuration? We should check that.

Reply:
I have to imagine there's a way to do it somehow - Or I wonder if maybe the K/N shims are doing the work of no-op'ing new APIs on older versions under the hood? I haven't looked into it tbh.